src/Xamarin.Android.Build.Utilities/Sdks/MonoDroidSdkUnix.cs

```diff
@@ -22,6 +22,9 @@ class MonoDroidSdkUnix : MonoDroidSdkBase
 		protected override string FindRuntime ()
 		{
 			string monoAndroidPath = Environment.GetEnvironmentVariable ("MONO_ANDROID_PATH");
+			if (monoAndroidPath == null)
+				// for Mono.Posix and Mono.Data.Sqlite builds in xamarin-android.
+				monoAndroidPath = Path.GetFullPath (Path.Combine (new Uri (GetType ().Assembly.CodeBase).LocalPath, "..", "..", "..", "..", ".."));
```

Comment:
Alright, I was confused about the formal(?) behaviour of the path probe, thought that user home thing is not kind of hack around some issue.